Add docs for configuring OpenTelemetry for self-hosted (#4255)
pulumi/pulumi-hugo: themes/default/content/docs/pulumi-cloud/self-hosted/components/api.md

The core infrastructure components to successfully run the API service are the database, object storage, and encryption services.
Depending on your requirements, you can configure additional (optional) identity services as well as enhanced security

> between the API and the database. The API also supports exporting OpenTelemetry metrics and traces via the OpenTelemetry collector.

Reviewer: ```suggestion
between the API and the database. The API also supports [exporting OpenTelemetry metrics and traces](#opentelemetry) via the OpenTelemetry collector.
```

      },
  ],
```

## OpenTelemetry

The API service is configured to export OpenTelemetry metrics and traces to the vendor of your choice via the OpenTelemetry collector. You will need to manage your own OpenTelemetry collector.

The following environment variables are needed to configure OpenTelemetry in the service:

| Variable name | Description |
| --- | --- |
| OTEL_EXPORTER_OTLP_ENDPOINT | (Required) Used to configure the OTLP exporter. The base URL to which all telemetry will be sent. If not set, the API service will use no-op metrics and traces. |
| OTEL_EXPORTER_OTLP_PROTOCOL | (Optional) Used to configure the OTLP exporter. Valid values are `http` or `grpc`. Defaults to `grpc`. |
| PULUMI_ENABLE_DEPRECATED_METRICS | (Optional) Whether to continue emitting API service metrics in a log-based format. Defaults to `true`. |
| METRICS_WEBHOOK_SECRET | Required to successfully authenticate to the `/metrics` endpoint. The Authorization header shoud be set as follows: `Authorization: webhook-token <METRICS_WEBHOOK_SECRET>`. |

> OpenTelemetry is not yet available for the node backend or console.

Reviewer: These are the same thing in terms of services, right? (Apologies if not -- it's been a while.) If so, should we just call this "the console service" (as that's what it seems to be called in the sibling docs)?

```suggestion
OpenTelemetry is not yet available for the [console service](/docs/pulumi-cloud/self-hosted/components/console/).
```

### Metrics endpoint

> The API service exposes a metrics endpoint (`https://api.pulumi.com/metrics`) that is secured by a bearer token. This token is configured using the environment variable, `METRICS_WEBHOOK_SECRET`.

Reviewer: ```suggestion
The API service exposes a metrics endpoint (`https://api.pulumi.com/metrics`) that is secured by a bearer token. This token is configured using the environment variable `METRICS_WEBHOOK_SECRET`.
```

{{% notes type="info" %}}
The token type is `webhook-token`, not `Bearer`.
{{% /notes %}}

```sh
curl -s GET https://api.pulumi.com/metrics -H 'Authorization: webhook-token <METRICS_WEBHOOK_SECRET>
```

### Prometheus

> The API service provides 2 options to get metrics into Prometheus:

Reviewer: ```suggestion
The API service provides two options to get metrics into Prometheus:
```

#### Prometheus remote write exporter

This option does not use the `/metrics` endpoint. Instead, it exports metrics from the collector to a [Prometheus remote write compatible backend](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusremotewriteexporter/README.md).

Example OpenTelemetry collector configuration for a service using AWS and the AWS Distro for OpenTelemetry Collector:

```yaml
extensions:
  sigv4auth:

receivers:
  otlp:
    protocols:
      grpc:
        endpoint: localhost:4317

processors:
  memory_limiter:

  batch:

exporters:
  logging:

  prometheusremotewrite:
    endpoint: https://aws-managed-prometheus-endpoint/v1/api/remote_write
    auth:
      authenticator: sigv4auth

service:
  telemetry:
    logs:

  pipelines:
    traces:
      receivers: [otlp]
      processors: [memory_limiter, batch]
      exporters: [logging]

    metrics:
      receivers: [otlp]
      processors: [memory_limiter, batch]
      exporters: [prometheusremotewrite]

  extensions: [sigv4auth]
```

#### Prometheus exporter

> This option require configuring the environment variable `METRICS_WEBHOOK_SECRET` to successfully authenticate to the [`/metrics` endpoint](#metrics-endpoint)

Reviewer: ```suggestion
This option requires configuring the environment variable `METRICS_WEBHOOK_SECRET` to successfully authenticate to the [`/metrics` endpoint](#metrics-endpoint).
```
